xbmc/threads/Condition.h

```diff
@@ -20,14 +20,59 @@
 
 #pragma once
 
-#include "threads/platform/Condition.h"
-
+#include "threads/SingleLock.h"
+#include "threads/Helpers.h"
 #include "threads/SystemClock.h"
-#include <stdio.h>
+
+#include <condition_variable>
+#include <chrono>
 
 namespace XbmcThreads
 {
 
+  /**
+   * This is a thin wrapper around boost::condition_variable. It is subject
+   *  to "spurious returns" as it is built on boost which is built on posix
+   *  on many of our platforms.
```

Comment:
comment is probably not valid anymore?

Reply:
It wasn't even valid before, boost is long gone :-)

Reply:
heh, "built on boost which is built on posix", conditions in posix can and will 'spurious return', you must check for it. The comment is quite valid but worded wrong now.

Reply:
Yeah. It's c-n-p from a platform specific file. I can fix it.